lib/services/credentials/credential.service.js

```diff
@@ -45,84 +45,94 @@ s.insertCredential = function (id, type, credentialDetails) {
     throw new Error(`Invalid credential type: ${type}`); // TODO: replace with validation error
   }
 
-  return this.getCredential(id, type) // check if credential already exists
-    .then((cred) => {
-      if (cred && cred.isActive) {
-        throw new Error('Credential already exists and is active'); // TODO: replace with validation error
-      }
-      return dereferencePromise;
-    })
-    .then((resolvedSchema) => {
-      const credentialConfig = resolvedSchema.properties[type];
-      const newCredential = { isActive: 'true' };
-      utils.appendCreatedAt(newCredential);
-      utils.appendUpdatedAt(newCredential);
+  // TODO: not a good approach, new way TBD
+  const areMultipleCredsAllowed = ['key-auth', 'jwt'].includes(type);
+
+  let credentialExistsPromise;
+  if (areMultipleCredsAllowed) {
+    credentialExistsPromise = Promise.resolve(dereferencePromise);
+  } else {
+    // check if credential already exists
+    credentialExistsPromise = this.getCredential(id, type)
+      .then(cred => {
+        if (cred && cred.isActive) {
+          throw new Error('Credential already exists and is active'); // TODO: replace with validation error
+        }
+        return dereferencePromise;
+      });
+  }
 
-      if (type === 'key-auth' || type === 'jwt') { // TODO: if/else is not a good approach, new way TBD
-        return Promise.all([
-          validateNewCredentialScopes(credentialConfig, credentialDetails),
-          validateNewCredentialProperties(credentialConfig, credentialDetails)
-        ]).then(([scopes, credentialProps]) => {
-          Object.assign(newCredential, credentialProps);
-          newCredential.keyId = credentialDetails.keyId || uuid62.v4();
-          newCredential.keySecret = credentialDetails.keySecret || uuid62.v4();
-          newCredential.scopes = JSON.stringify(scopes);
-          newCredential.consumerId = id;
-
-          return Promise.all([
-            credentialDao.insertCredential(newCredential.keyId, type, newCredential),
-            credentialDao.associateCredentialWithScopes(newCredential.keyId, type, scopes)
-          ]);
-        }).then(() => {
-          if (newCredential.scopes && newCredential.scopes.length > 0) {
-            newCredential.scopes = JSON.parse(newCredential.scopes);
-          }
-
-          if (typeof newCredential.isActive === 'string') {
-            newCredential.isActive = newCredential.isActive === 'true';
-          }
-          return newCredential;
-        });
-      }
+  return credentialExistsPromise.then((resolvedSchema) => {
```

Comment:
If it's a single param we can remove parenthesis
```suggestion
  return credentialExistsPromise.then(resolvedSchema => {
```